tests/func/experiments/test_queue.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+from funcy import first
+
+from dvc.exceptions import InvalidArgumentError
+
+
+def to_dict(tasks):
+    status_dict = {}
+    for task in tasks:
+        status_dict[task["name"]] = task["status"]
+    return status_dict
+
+
+@pytest.fixture
+def queued_tasks(tmp_dir, dvc, scm, exp_stage):
+    queue_length = 3
+    name_list = []
+    for i in range(queue_length):
+        name = f"queued{i}"
+        name_list.append(name)
+        dvc.experiments.run(
+            exp_stage.addressing,
+            params=[f"foo={i+2*queue_length}"],
+            queue=True,
+            name=name,
+        )
+    return ["queued0", "queued1", "queued2"]
+
+
+@pytest.fixture
+def success_tasks(tmp_dir, dvc, scm, test_queue, exp_stage):
+    queue_length = 3
+    name_list = []
+    for i in range(queue_length):
+        name = f"success{i}"
+        name_list.append(name)
+        dvc.experiments.run(
+            exp_stage.addressing, params=[f"foo={i}"], queue=True, name=name
+        )
+    dvc.experiments.run(run_all=True)
+    return name_list
+
+
+@pytest.fixture
+def failed_tasks(tmp_dir, dvc, scm, test_queue, failed_exp_stage, capsys):
+    queue_length = 3
+    name_list = []
+    for i in range(queue_length):
+        name = f"failed{i}"
+        name_list.append(name)
+        dvc.experiments.run(
+            failed_exp_stage.addressing,
+            params=[f"foo={i+queue_length}"],
+            queue=True,
+            name=name,
+        )
+    dvc.experiments.run(run_all=True)
+    output = capsys.readouterr()
+    assert "Failed to reproduce experiment" in output.err
+    return name_list
```

Comment:
Fixtures should not contain tests themselves, they should only provide any necessary setup and/or cleanup needing to run tests that use the fixture.

It's general it would be ok to have a test fixture that sets up some set of failed experiments, but `test_failed_exp_celery` should be separate from that. It does not make sense to have a repeated UI test that runs every time we re-use that fixture.

However, I think that you probably won't be able to use fixtures that run celery experiments in this way. We will end up hitting race conditions due to how the pytest-celery worker executes tasks.

Basically it will be safer to not use fixtures that run experiments, and setup everything in individual tests.